stuff/pipelines/client_contest.py

```diff
@@ -0,0 +1,162 @@
+        queue=queue_name,
+        routing_key="ContestQueue")
+    logging.info('[*] waiting for the messages')
+
+    def callback(_channel: pika.adapters.blocking_connection.BlockingChannel,
+                 _method: pika.spec.Basic.Deliver,
+                 _properties: pika.spec.BasicProperties,
+                 body: bytes) -> None:
+        data = json.loads(body.decode())
+        cur.execute('''
+                SELECT
+                    @n := @n + 1 place,
+                    r.contest_score,
+                    r.username,
+                    r.identity_id,
+                    c.certificate_cutoff,
+                    ce.certificate_id
+                FROM
+                    (
+                        SELECT
+                            SUM(r.contest_score) AS contest_score,
+                            i.username,
+                            i.identity_id,
+                            pp.problemset_id
+                        FROM
+                            Problemset_Problems pp
+                        INNER JOIN
+                            Submissions s
+                        ON
+                            s.problemset_id = pp.problemset_id
+                        INNER JOIN
+                            Runs r
+                        ON
+                            s.current_run_id = r.run_id
+                        INNER JOIN
+                            Identities i
+                        ON
+                            i.identity_id = s.identity_id
+                        INNER JOIN
+                            Contests co
+                        ON
+                            pp.problemset_id = co.problemset_id
+                        WHERE
+                            co.contest_id = %s
+                            AND r.status = 'ready'
+                            AND s.type = 'normal'
+                            AND r.verdict NOT IN ('CE', 'JE', 'VE')
+                        GROUP BY
+                            s.identity_id
+                        ORDER BY
+                            contest_score DESC
+                    ) AS r
+                INNER JOIN
+                    Contests c
+                ON
+                    c.problemset_id = r.problemset_id
+                LEFT JOIN
+                    Certificates ce
+                ON
+                    r.identity_id = ce.identity_id
+                    AND c.contest_id = ce.contest_id
+                CROSS JOIN
+                    (SELECT @n := 0) AS temp
+                WHERE
+                    certificate_id IS NULL;
+                ''', (data['contest_id'],))
+        certificates = []
+        for row in cur:
+            contest_place = None
+            if row['certificate_cutoff'] is None:
+                logging.info('The contest has no places')
+            elif row['place'] > row['certificate_cutoff']:
+                logging.info('The user did not reach the place to be reported')
```

Comment:
una vez que se llegue aquí, se puede hacer

```suggestion
                break
```

guard clause pattern ftw!

Reply:
De acuerdo a lo descrito en el documento de diseño:
```
`certificate_cutoff`:  Entero que indicará el número de concursantes a premiar 
con diplomas (por ejemplo, si certificate_cutoff = 3, entonces se le otorgará 
un diploma con mención de lugar en el ranking únicamente a los 3 estudiantes 
con mejor puntaje; a los demás se les otorgará un diploma de participación que
no menciona el lugar).
```

También se debería generar diploma de participación a los concursantes que no alcanzaron la posición configurada. Así que si agregamos supongo que ese break no nos permitiría insertar los registros para ellos.

Reply:
oh ya, entonces se puede hacer mejor

```python
contest_place: Optional[int] = None
if data['certificate_cutoff'] and row['place'] <= data['certificate_cutoff']:
  contest_place = row['place']
```

los logs son muy verbosos ^^;;

Reply:
Listo. Aún con la duda de si los concursos que no tienen un valor en `certificate_cutoff` vayan a entregar diploma de participación o no. 

El cambio sería mínimo:
```py3
contest_place: Optional[int] = None
if data['certificate_cutoff'] is None:
  break
if row['place'] <= data['certificate_cutoff']:
  contest_place = row['place']
```